src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
> I thought about this a bit as well and I think the handler separation is a good thing at the moment --- as stampeding puppet agents are the problem we want to target and rejecting other traffic such as console or API requests would be a problem. It might be worth enforcing this by adding a check for the Puppet User-Agent header.

Yeah, I suppose with the QoSFilter-style approach, we could prioritize those ahead of the other agent-based ones to make it much less likely that those would be dropped but what you have for now in lieu of that probably makes sense.  I wouldn't object to an extra `User-Agent` header check here if you want to add that but not sure it's essential at this point.  Maybe simpler is better for now.